File: _BHCVRP_Python_Version/cujae_inf_citi_om/data/Depot.py
from data import Location

# Clase que modela los datos de un depósito en un VRP

class Depot:
    
    def __init__(self, id_depot, location_depot: Location, list_fleets=None):
        self._id_depot = id_depot
        self._location_depot = location_depot
        self._list_fleets = list_fleets if list_fleets is not None else []

    def get_id_depot(self):
        return self._id_depot

    def set_id_depot(self, id_depot):
        self._id_depot = id_depot
        
    def get_list_fleets(self):
        return self._list_fleets

    def set_list_fleets(self, list_fleets):
        self._list_fleets = list_fleets

    def get_location_depot(self):
        return self._location_depot

    def set_location_depot(self, location_depot: Location):
        self._location_depot = location_depot

    

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/data/Depot.java
package cujae.inf.citi.om.data;

import java.util.ArrayList;

/* Clase que modela los datos de un depósito en un VRP*/

public class Depot {
	
	protected int idDepot;
	protected Location locationDepot;
	protected ArrayList<Fleet> listFleets;

	public Depot() {
		super();
		listFleets = new ArrayList<Fleet>();
	}
	 
	public Depot(int idDepot, Location locationDepot, ArrayList<Fleet> listFleets) {
		super();
		this.idDepot = idDepot;
		this.locationDepot = locationDepot;
		this.listFleets = listFleets;
	}

	public ArrayList<Fleet> getListFleets() {
		return listFleets;
	}

	public void setListFleets(ArrayList<Fleet> listFleets) {
		this.listFleets = listFleets;
	}

	public Location getLocationDepot() {
		return locationDepot;
	}

	public void setLocationDepot(Location locationDepot) {
		this.locationDepot = locationDepot;
	}

	public int getIdDepot() {
		return idDepot;
	}

	public void setIdDepot(int idDepot) {
		this.idDepot = idDepot;
	}
}
